### ISSUE TITLE
page_rotation_reset does not return page to original rotation

### ISSUE THREAD

**Issue by MaryiaTsalapava** [pymupdf/PyMuPDF#3062]
### Description of the bug

Original document where pages were rotated is changed after running `find_tables`.
`page_rotation_reset` does not return page to original state.

### How to reproduce the bug

Steps to reproduce:
- Find tables on page, that is rotated. Page rotation is 90 degrees in the example.
- Save current document
- Page has changed rotation in the original document

Expected behavior:
- Original document page is not changed after finding tables
- `page_rotation_reset` returns page to original state

Actual behavior:
- Original document page is rotated after finding tables
- `page_rotation_reset` does not return page to original state
- Looks like the rotated page is the result page from `page_rotation_set0`

```
import fitz
from fitz.table import find_tables


doc = fitz.Document("sample.pdf")
for page in doc.pages():
    for table in find_tables(page):
        for cell in table.cells:
            rect = fitz.Rect(*cell)
            highlight = page.add_rect_annot(rect)
            highlight.update()

doc.save("sample_output.pdf")

```

Sample document and document with changed page are attached:
[sample.pdf](https://github.com/pymupdf/PyMuPDF/files/13979410/sample.pdf)
[sample_output.pdf](https://github.com/pymupdf/PyMuPDF/files/13979417/sample_output.pdf)

### PyMuPDF version

1.23.15

### Operating system

Windows

### Python version

3.9

**Comment by MaryiaTsalapava** [comment]
Ok, thank you

**Comment by JorjMcKie** [comment]
Thank you very much for the confirmation!

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.17.
